src/query/pipeline/sources/src/processors/sources/input_formats/input_pipeline.rs

```diff
@@ -203,13 +204,28 @@ pub trait InputFormatPipe: Sized + Send + 'static {
                 let splits = splits.to_owned().clone();
                 tokio::spawn(async move {
                     let mut futs = FuturesUnordered::new();
-                    for s in &splits {
-                        let fut = Self::read_split(ctx_clone2.clone(), s);
+                    for s in splits.into_iter() {
+                        let fut =
+                            CatchUnwindFuture::create(Self::read_split(ctx_clone2.clone(), s));
                         futs.push(fut);
                     }
                     while let Some(row_batch) = futs.next().await {
-                        if data_tx2.send(row_batch.unwrap()).await.is_err() {
-                            info!("execute_copy_aligned fail to send row_batch")
+                        match row_batch {
+                            Ok(row_batch) => {
+                                let is_err = row_batch.is_err();
+                                if data_tx2.send(row_batch).await.is_err() {
+                                    info!("execute_copy_aligned fail to send row_batch")
```

Comment:
Why use info instead of error here?

Reply:
most of the time, if the receiver closes first，the processor in the receiver side may encounter error, 
so we in fact do not need the log.  logging it to help the debug.

channel is a workaroud, later we will read in pipeline,  and not need the channel any more.

Reply:
This info log may flood the log, please remove it.

Reply:
> most of the time, if the receiver closes first，the processor in the receiver side may encounter error, so we in fact do not need the log. logging it to help the debug.
> 
> channel is a workaroud, later we will read in pipeline, and not need the channel any more.

This log is not useful here, it looks should be an error and log the error msg.

Reply:
done